core/server/worker/src/main/java/alluxio/worker/page/PagedBlockMetaStore.java

```diff
@@ -404,6 +404,21 @@ public PagedBlockStoreMeta getStoreMetaFull() {
         usedBytesOnDirs.build(), blockOnDirs.build());
   }
 
+  @Override
+  @GuardedBy("getLock().readLock()")
+  public void reportBlocks(PageStoreDir pageStoreDir) {
+    final PagedBlockStoreDir pagedBlockStoreDir = downcast(pageStoreDir);
+    Set<PagedBlockMeta> blockMetas = mBlocks.getByField(INDEX_STORE_DIR, pagedBlockStoreDir);
+    for (PagedBlockMeta blockMeta : blockMetas) {
+      for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
+        synchronized (listener) {
+          listener.onMoveBlockByWorker(blockMeta.getBlockId(),
```

Comment:
This is a tricky way to report the blocks generated by pageStore to master by Incrementally, but it need a full block report during register stage.